server/ntb/publish/ntb_nitf.py

```diff
@@ -73,6 +76,31 @@ def format(self, article, subscriber, codes=None, encoding="us-ascii"):
         except Exception as ex:
             raise FormatterError.nitfFormatterError(ex, subscriber)
 
+    def _populate_metadata(self, article):
+        """
+        For tree type vocabularies add the parent if a child is present
+        """
+        vocabularies = list(superdesk.get_resource_service('vocabularies').get(None, None))
+        fields = {'place': 'place_custom', 'subject': 'subject_custom'}
+        for field in fields:
+            vocabulary = self._get_list_element(vocabularies, '_id', fields[field])
+            if not vocabulary:
+                continue
+            vocabulary_items = vocabulary.get('items', [])
+            field_values = article.get(field, [])
+            for value in list(field_values):
+                if value.get('parent', None) and not self._get_list_element(field_values, 'qcode', value['parent']):
```

Comment:
Will be better to use one more variable for: `self._get_list_element(field_values, 'qcode', value['parent'])`